client/pdf_generator.py

```diff
@@ -0,0 +1,186 @@
+
+try:
+	from reportlab import platypus
+	from reportlab.lib import styles
+	from reportlab.lib.enums import TA_JUSTIFY
+	from reportlab.lib.pagesizes import letter
+	from reportlab.lib.units import inch
+except ImportError:
+	has_reportlab = False
+else:
+	has_reportlab = True
+
+def _expand_path(outfile, *joins, pathmod=os.path):
+	outfile = pathmod.expandvars(outfile)
+	outfile = pathmod.expanduser(outfile)
+	outfile.join(outfile, *joins)
+	return outfile
+
+class Plugin(plugins.ClientPlugin):
+	authors = ['Jeremy Schoeneman']
+	title = 'Generate PDF'
+	description = """
+	Generates a PDF file with a link which includes the campaign url with the
+	individual message_id required to track individual visits to a website.
+	Visit https://github.com/y4utj4/pdf_generator for example template files to
+	use for this plugin.
+	"""
+	homepage = 'https://github.com/securestate/king-phisher-plugins'
+	options = [
+		plugins.ClientOptionString(
+			'output_pdf',
+			'pdf being generated',
+			default='~/Attachment1.pdf',
+			display_name='* Output PDF File'
+		),
+		plugins.ClientOptionString(
+			'template_file',
+			'Template file to read from',
+			default='~/template.txt',
+			display_name='* Template File'
+		),
+		plugins.ClientOptionString(
+			'logo',
+			'Image to include into the pdf',
+			display_name='Logo / Inline Image'
+		),
+		plugins.ClientOptionString(
+			'link_text',
+			'Text for inserted link',
+			default='Click here to accept',
+			display_name='Link Text'
+		)
+	]
+	req_min_version = '1.7.0b1'
+	req_packages = {
+		'reportlab': has_reportlab
+	}
+	def initialize(self):
+		mailer_tab = self.application.main_tabs['mailer']
+		self.text_insert = mailer_tab.tabs['send_messages'].text_insert
+		self.add_menu_item('Tools > Create PDF Preview', self.make_preview)
+		self.signal_connect('send-precheck', self.signal_send_precheck, gobject=mailer_tab)
+		self.signal_connect('send-target', self.signal_send_target, gobject=mailer_tab)
+		self.signal_connect('send-finished', self.signal_send_finished, gobject=mailer_tab)
+		return True
+
+	def signal_send_precheck(self, _):
+		# file and option checks
+		if not all((self.config['template_file'], self.config['output_pdf'], self.config['link_text'])):
+			self.logger.debug('skipping exporting any attachments due to lack of information provided to run')
+			return 0
+		if not os.path.isfile(self.config['template_file']):
+			self.logger.error('template file does not exist, specify a valid template file')
+			return 0
+		if self.config['logo'] and not os.path.isfile(self.config['template_file']):
+			self.logger.error('specified template or logo file not found')
+		self.logger.debug('pdf template file found, generating attachment')
+		return True
+
+	def make_preview(self, _):
+		if not os.path.isfile(self.config['template_file']):
+			self.logger.error('template file does not exist, please review your options')
+			return 0
+		else:
+			outfile = self.expand_path(self.config['output_pdf'])
```

Comment:
This entire else block could be removed and de-indented because the conditional block above ends with a return statement.